components/playlist/playlist_download_request_manager.cc

```diff
@@ -153,11 +153,19 @@ void PlaylistDownloadRequestManager::GetMedia(content::WebContents* contents) {
   DVLOG(2) << __func__;
   DCHECK(contents && contents->GetPrimaryMainFrame());
 
+#if BUILDFLAG(IS_ANDROID)
+  content::RenderFrameHost::AllowInjectingJavaScript();
```

Comment:
> somewhere that the frame is a main frame and the origin is

Not now, but I think I can.


> do you understand the security consequences of setting this?

Not for sure, but I'd like to mention that it seems we're already doing a similar thing in background_video_playback_tab_helper.

Reply:
> Not now, but I think I can.

actually this check probably doesn't make sense since it's a global setting

Reply:
> Not now, but I think I can.

Ah, yeah. I misunderstood about your question. The setting is global.

Reply:
@sangwoo108 correct me if I'm wrong, the purpose of this injection is to evaluate a JS in the context of the page (m.youtube.com) and pass the result (video details, such as name, author, etc) of evaluation back to the browser. As per my understanding the JS we inject is only executed in the context of the page and not a secure context of the browser, so technically a page can pollute the prototypes, it also can achieve the same result by just changing its JS objects (it can put anything into `window.ytplayer?.bootstrapPlayerResponse?.videoDetails?.author`, it doesn't need to work with prototypes). If we can't implement the same mechanism with mutation observers that we have in iOS I recommend do not spend time on making it resisted to prototype pollution, but instead check for the data we pass from the page to the browser as @diracdeltas mentioned here: https://github.com/brave/brave-core/pull/16419/files#r1055889333

It's important to check in the native code and not in JS obviously, otherwise we have the same issue with the pollution.

Reply:
> thee purpose of this injection is to evaluate a JS in the context of the page (m.youtube.com) and pass the result (video details, such as name, author, etc) of evaluation back to the browser.

Yes, it's correct. But this script will be executed not only for `m.youtube.com` but also arbitrary pages.

>  If we can't implement the same mechanism with mutation observers

Unfortunately, we can't. We couldn't find out a way to fire an event from injected script to C++.